### PR TITLE
Fit tzrmjd

### PR DESCRIPTION
**Pull request by luojing1211**
This is a pull request for fixing issue #431.

This branch is still under development.

### CODE REVIEW

**Comment by scottransom** [comment]
What is the status of this PR?  I have a student, @clp3ef (Camryn Phillips) who is working on automatic phase connection (and random model generation) who needs this functionality.  I'll send an email summarizing what she is doing to the mailing list.

**Comment by luojing1211** [comment]
We have not decided how to handle the " d_abs_phase_d_F0 " problem. While doing fitting, should we vary F0 in the absolute phase model?

**Comment by scottransom** [comment]
@luojing1211 I can't remember what the problem actually is...  Can you summarize?

**Comment by luojing1211** [comment]
In the current setup. 

```
result_Phase(TOA) = Phase(TOA) - Phase(TZRMJD) 
```
However, for instance, when we doing the fit for F0. The design matrix will be: 
```
d_result_Phase(TOA)/d_F0 = d_Phase(TOA) /d_F0 - d_Phase(TZRMJD)/d_F0 
```
Before, 
```
result_Phase(TOA) = Phase(TOA -TZRMJD),
```
 the derivative is:
```
d_result_Phase(TOA)/d_F0 = d_Phase(TOA - TZRMJD) /d_F0
```
We have not decide which one is should be the correct answer, or what is the correct way to do it?

**Comment by abhisrkckl** [comment]
Related PR #1557
